src/scrapers/leumi-card.js

```diff
@@ -286,17 +287,18 @@ async function fetchTransactions(browser, options, navigateToFunc) {
 
   const allTasks = [];
   for (let i = 0; i < allMonths.length; i += 1) {
-    const task = fetchTransactionsForMonth(browser, navigateToFunc, allMonths[i]);
+    const task = () => fetchTransactionsForMonth(browser, navigateToFunc, allMonths[i]);
     allTasks.push(task);
   }
 
-  const task = fetchTransactionsForMonth(browser, navigateToFunc);
+  const task = () => fetchTransactionsForMonth(browser, navigateToFunc);
   allTasks.push(task);
 
-  const allTasksResults = await Promise.all(allTasks);
-  const allResults = allTasksResults.reduce((obj, result) => {
-    return addResult(obj, result);
-  }, {});
+  let allResults = {};
+  for (const task of allTasks) {
```

Comment:
@esakal if we cannot fetch results asynchronously, we might as well fetch them within the loops 👆 
(so no need to create an array of tasks).

Reply:
sure, I tried to keep changes to minimum but I agree it will simplify the logic.

btw, although before this PR we opened all browsers in parallel, we still scraped each one of them in a sequence thus I don't expect a noticeable performance hit

Reply:
👍